question_classifier.py

```diff
@@ -144,6 +148,53 @@ def build_question_classifier(self):
         self.classifier = sklearn.neighbors.KNeighborsClassifier(n_neighbors=1)
         self.classifier.fit(vectors, y_train)
 
+    def filterWHTags(self, question):
+        # ADD ALL VARIABLES TO THE FEATURE DICT WITH A WEIGHT OF 90
+        matches = re.findall(r'(\[(.*?)\])', question)
+        for match in matches:
+            question = question.replace(match[0], '')
+
+        question = re.sub('[^a-zA-Z0-9]', ' ', question)
+
+        # PRE-PROCESSING: TOKENIZE SENTENCE, AND LOWER AND STEM EACH WORD
+        words = nltk.word_tokenize(question)
+        words = [word.lower() for word in words if '[' and ']' not in word]
+
+        filtered_words = self.get_lemmas(words)
+
+        question_tags = nltk.pos_tag(filtered_words)
+        question_tags = [
+            tag for tag in question_tags if self.is_wh_word(tag[1])]
+        return question_tags
+
+    def validate_WH(self, test_question, predicted_question):
+        """
+        Assumes that only 1 WH word exists
+        Returns True if the WH word in the test question equals the
+        WH word in the predicted question
+        """
+
+        test_tags = self.filterWHTags(test_question)
+        predicted_tags = self.filterWHTags(predicted_question)
+
+        # Uncomment these lines below to see
+        # print("Test")
+        # print(test_tags)
+        # print()
+
+        # print("Predicted")
+        # print(predicted_tags)
+        # print()
+
+        # Compares all WH words in the tags array and returns False if one doesn't match
+        min_tag_len = min(len(test_tags), len(predicted_tags))
+        wh_match = True
+        i = 0
+        while (wh_match and i < min_tag_len):
+            wh_match = wh_match and (test_tags[0][i] == predicted_tags[0][i])
```

Comment:
I think the indexing here needs to be reversed, `test_tags[i][0] == predicted_tags[i][0]`, Since `test_tags = [("word", PART_OF_SPEECH),...]`.

Reply:
Yup, you're right, good catch!